x-pack/filebeat/input/awss3/sqs.go

```diff
@@ -76,37 +158,88 @@ func (r *sqsReader) Receive(ctx context.Context) error {
 		// Release unused workers.
 		r.workerSem.Release(workers - len(msgs))
 
-		// Process each SQS message asynchronously with a goroutine.
 		r.log.Debugf("Received %v SQS messages.", len(msgs))
 		r.metrics.sqsMessagesReceivedTotal.Add(uint64(len(msgs)))
-		workerWg.Add(len(msgs))
+
+		// Add to processing wait group to wait for all messages to be processed.
+		processingWg.Add(len(msgs))
```

Comment:
Why do we still need an artificial messages limit? Why can't we just process objects and messages as fast as they are available?

Reply:
It's not a message, limit, it is a limit on the size of the messages batch we process concurrently. We move to the next batch after all events from the batch are sent to the queue.
We could not wait for the last, I will run a benchmark scenario, but I expect goroutines to increase.

Reply:
Ah got it, the meaning of "max_number_of_messages" has changed to be more like a concurrency limit. Once we finalize things we'll have to update the docs to explain this.